compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp

```diff
@@ -476,6 +476,27 @@ static void propagateLayoutToTransposeOp(
   update(result, changed);
 }
 
+static void propagateLayoutToContractionOp(
+    vector::ContractionOp contraction,
+    ArrayRef<const DistributionLayout *> operandLattices,
+    ArrayRef<DistributionLayout *> resultLattices,
+    std::function<void(DistributionLayout *, ChangeResult)> update) {
+  // Contraction has only one vector result.
+  DistributionLayout *result = resultLattices[0];
+  // Get the init value of the contraction.
+  const DistributionLayout *init = operandLattices[2];
```

Comment:
What happens if the init value does not have a layout?

Reply:
It's already handeled by resolve. Resolve will not resolve if the layout doesn't exist.